source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -277,7 +277,37 @@ ShaderPtr OslShaderGenerator::generate(const string& name, ElementPtr element, G
 
     // Emit shader output
     const VariableBlock& outputs = stage.getOutputBlock(OSL::OUTPUTS);
-    emitShaderOutputs(outputs, stage);
+    const ShaderPort* singleOutput = outputs.size() == 1 ? outputs[0] : NULL;
+
+    const bool isSurfaceShaderOutput = singleOutput && singleOutput->getType() == Type::SURFACESHADER;
+
+#ifdef MATERIALX_OSL_LEGACY_CLOSURES
+    const bool isBsdfOutput = singleOutput && singleOutput->getType() == Type::BSDF;
+#endif
+
+    if (isSurfaceShaderOutput)
```

Comment:
This logic added here is a bit hacky with hard coded handling of `surfaceshader` and `BSDF` as output types. The reason for this is to keep supporting output of nodegraphs / documents that does not have a material node as the final output terminal (there are many exampels of this in out test suite).

For now this is hard coded here for `surfaceshader` and `BSDF` types. I have a plan for how to improve and generalize this, but it includes change to our `Syntax` and `TypeSyntax` classes that I think is too intrusive for 1.38.6. So I'll hold on this until 1.39.